modules/javafx.web/src/main/native/Source/WTF/wtf/java/MainThreadJava.cpp

```diff
@@ -30,11 +30,20 @@
 #include <wtf/MainThread.h>
 #include <wtf/RunLoop.h>
 
+#if OS(UNIX)
+#include <pthread.h>
+#endif
+
 namespace WTF {
 static JGClass jMainThreadCls;
-static jmethodID fwkIsMainThread;
 static jmethodID fwkScheduleDispatchFunctions;
 
+#if OS(UNIX)
+static pthread_t mainThread;
+#elif OS(WINDOWS)
+static ThreadIdentifier mainThread { 0 };
+#endif
```

Comment:
Both here and below, where the `isMainThread` is defined, assume that `UNIX` and `WINDOWS` are the only two options. I presume this is a valid assumption?

Reply:
I've tested in all three platforms. The two options are sufficient as Mac and Linux uses `pthread`.

Reply:
That's what I figured. It's only theoretical, in that if we ever add a platform that isn't one of `WINDOWS` or `UNIX` it wouldn't compile, but I imagine we have very many other places where that is also true.